Site Selector: Use text rather than icons for flags in the sidebar for clarity

sixhours opened the pull request:
#### Changes proposed in this Pull Request

* Same changes as in #35241 but this also updates the other types of icons (Redirect and Domain Only) to use text rather than an icon for clarity.
* I'm not clear about what a "domain-only" site is, so I'm not sure how we should refer to it in text, or how to make that appear for testing purposes.

**Before**

<img width="273" alt="Screen Shot 2019-09-24 at 12 18 37 PM" src="https://user-images.githubusercontent.com/2124984/65530210-7edf5580-dec5-11e9-8a5a-33ebe81665a9.png">

**After**

<img width="273" alt="Screen Shot 2019-09-24 at 12 17 59 PM" src="https://user-images.githubusercontent.com/2124984/65530232-87379080-dec5-11e9-9d53-bed800e961ee.png">

#### Testing instructions

* Switch to this PR
* Check the sidebar on a site that is private; you should see the above site privacy flag next to the domain.
* Check sites that are only used as redirects to ensure you see the "Redirect" text and style rather than an icon
* I don't know how to get the domain-only flag to appear.
* Check other color schemes to ensure there's sufficient contrast.
* Check other instances of the site selector (Write button in the masterbar, site selector when viewing a URL that has no site specified) to ensure they appear as expected.

sixhours committed:
Use text rather than icons for flags in the sidebar.

a8ci18n commented on client/blocks/site/index.jsx:
```diff
@@ -146,18 +146,16 @@ class Site extends React.Component {
-								<span className="site__badge">
-									<Gridicon icon="block" size={ 14 } />
+								<span className="site__badge site__badge-redirect">
+									{ translate( 'Redirect' ) }
```
Hi! I've found a possible matching string that has already been translated **1** time:
`translate( 'Redirect', { context: 'post type singular name'} )` *ES Score: 13*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

jsnajdr commented:
One thing that worries me here: the text badge pushes the site title text to the right, making only a much smaller part of it visible.

When localized to other languages, the translated label can be much longer than the English word, possibly making the site title completely invisible.

An alternative solution would be to keep the icon, but attach a tooltip to it, so that the user has a chance to figure out what the icon means.

sixhours commented:
@shaunandrews @michaeldcain Something to consider in @jsnajdr's comment above. ^ The pill-button style could cause the text to run long in other languages and hide the site URL. Thoughts on how to account for this in the design?

shaunandrews commented:
Perhaps we drop the badge down to a new line:

<img width="296" alt="image" src="https://user-images.githubusercontent.com/191598/64128667-a0956300-cd85-11e9-9d3a-7a5142df7993.png">

shaunandrews commented:
<img width="285" alt="image" src="https://user-images.githubusercontent.com/191598/64128871-d5ee8080-cd86-11e9-9b72-5228b67a62c3.png">

Worth mentioning that there are sometimes icons inserted on the right to indicate other states.

jsnajdr commented:
> Perhaps we drop the badge down to a new line:

That makes the `Site` item variable-height, doesn't it? Not cools for lists 🙂

shaunandrews commented:
> That makes the Site item variable-height, doesn't it? Not cools for lists

Why?

donpark commented:
Consider lowercasing tags. I think we just want to distinguish in this tag use-case and CamelCase goes tad beyond that.

sixhours committed:
Update colors with new vars.

sixhours committed:
Move site flag underneath domain for better readability.

sixhours committed:
Add cleaner hover styles for the site flag so it doesn't blend in with the background color on hover.

sixhours commented:
Moved the pill button flags underneath; this doesn't seem to break anything. Also updated the color values to reflect the new color palette work, and added inverse hover styles to avoid the pill blending into the background on hover.

jancavan commented:
> Not cools for lists

I think because if height is variable then there's a possibility that each list would have differing heights? But I think we restrict the site title, URL (and badge?) to a single line so the site item will always be the same height.

shaunandrews commented:
> each list would have differing heights

I’m not sure how this is a problem?

jancavan commented:
> I’m not sure how this is a problem?

I'm asking also, I thought maybe visually it'll look off in this particular context, but that was also just a guess.

jsnajdr commented:
> I’m not sure how this is a problem?

Things that have a regular structure are generally more pleasant to look at, that's all 🙂 I don't have any stronger argument.

shaunandrews commented:
I see the difference in height as another indicator that one site is different from the others; it helps to further highlight that a site is private or a domain only, etc.

p-jackson reviewed (commented):
Nice @sixhours, I personally find the text labels much easier to understand.

Code looking good, I've added e2e tests to make sure the DOM changes don't break any existing tests.
Desktop build breakage is a known issue from a while ago, but I believe would be fixed by a rebase.

* The colour inverting styles work well in the sidebar but look a bit strange on the `/me/purchases` page.
  ![Oct-02-2019 12-15-02](https://user-images.githubusercontent.com/1500769/66007366-97c3b880-e50e-11e9-9af7-187d4f771cf5.gif)

* Maybe there should be some extra space where there are two badges side-by-side. If you need to create a redirect site for testing you can [follow these instructions](https://en.support.wordpress.com/site-redirect/).

* Strawman example (my sidebar is a mess) but things get jumbled together in a large sidebar. It's hard to know which site a label is associated with (I don't think padding differences are always easy for non-designers to see). It was nice how the text used to fit neatly within the vertical bounds of the site icon. On balance though I'd much rather have more readable labels than the icons that didn't have this issue.
  <img width="231" alt="Screenshot 2019-10-02 at 12 26 48 PM" src="https://user-images.githubusercontent.com/1500769/66007703-efaeef00-e50f-11e9-8b36-9cb257a99c70.png">

jancavan commented:
> It's hard to know which site a label is associated with (I don't think padding differences are always easy for non-designers to see)

Maybe divider lines can work here.

sixhours committed:
Add margin so sites with multiple flags don't run together.

sixhours committed:
Make hover styles more specific, and exclude certain situations from hover styles.

sixhours commented:
> Maybe divider lines can work here.

I think a super-subtle line might help, yes:

<img width="304" alt="Screen Shot 2019-10-02 at 12 05 26 PM" src="https://user-images.githubusercontent.com/2124984/66061264-1f102c80-e50d-11e9-9fec-7a49d764809c.png">

That said, I think we'd be OK going without, too. IIRC we don't have too many users with multiple sites (not at the level of most Automatticians :D) so this feels stress-case-y to me. 

Thanks much for the review @p-jackson ! I've made some fixes based on your feedback. Let me know what you think about adding a separator like the above.

donpark commented:
Why not just add a line for tags in all items? They can be used to display site status, domain/plan expiration, etc.

shaunandrews commented:
I don't mind the line, but don't have a strong opinion.

> we don't have too many users with multiple sites (not at the level of most Automatticians :D) so this feels stress-case-y to me.

Very important to remember this. I wouldn't hold this up any longer — 🚢

p-jackson reviewed (approved):
Re-checked with the changes to the padding and hover rules. LGTM!

I thought that those extra commits would have addressed the issue with that `ci/circleci: wait-calypso-live` check. Not sure what's going wrong there.
